### PR TITLE
[feature]: Header Component

### PR DESCRIPTION
**Pull request by Himanshushakyaa**
## Description

-   [Ticket Link][NA]
-   Make a header fully responsive .

## UI screenshot or Video

chrome_UgPF4EkEk8

---

### General:

-   [x] **PR Size :** Confirming PR size < 20 files or pre-approved if > 20 files.
-   [x] **Self Review :** Confirming I have self-reviewed my PR
-   [x] **Peers Review :** - [ ] @Dev 1
-   [x] Confirming I have added all sensitive information-related should be inside .env files

### Client:

-   [x] **Console :** Confirming no console errors in the UI for this update.

-   [x] **Import order :** Confirming correct import order.

```
# External Import
import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';

# Internal Import
import { } from '../../Api/ApiRoutes';
import { } from '../../constants/AppConstant';
```

-   [ ] **Re-renders :** Confirming minimal re-renders for new React components.

-   [ ] **Network calls :** Confirming efficient network/API calls, minimized as needed.

-   [ ] **Util function comments :** Confirming comprehensive comments for utility functions.

-   [x] **Image optimization :** Confirming I have checked that all images are optimized. If not, please use https://tinyjpg.com/ or https://vecta.io/nano.

-   [x] **Dead code :** Confirming I have removed all comments code which is of no use OR written why I have kept the commented code.

### Confirming different viewport testing of the entire flow:

-   [x] Tested on 280px viewport
-   [x] Tested on 320px viewport
-   [x] Tested on 375px viewport
-   [x] Tested on 425px viewport
-   [x] Tested on 768px viewport
-   [x] Tested on 1024px viewport
-   [x] Tested on 1440px viewport
-   [x] Tested on 2560px viewport

### End-to-end testing:

-   [x] **Happy flow :** Successfully tested end-to-end happy flow, all working as expected.
-   [ ] **Error flow :** Tested UI flow with error scenarios, gracefully handled errors on UI
-   [ ] **Common component changed :** Tested change in common component across views to prevent breaking elsewhere.

### CODE REVIEW

File: src/components/header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  return (
+    <div className="fixed top-0 w-full bg-white z-50  ">
+      <div className=" container mx-auto px-3">
+        <div className="flex justify-between  items-center py-4">
+          <div className="flex  gap-2 md:gap-8  justify-center items-center">
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: not formatted

File: src/components/header/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  return (
+    <div className="fixed top-0 w-full bg-white z-50  ">
+      <div className=" container mx-auto px-3">
+        <div className="flex justify-between  items-center py-4">
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: not formatted

File: src/components/header/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          <div className="flex  gap-2 md:gap-8  justify-center items-center">
+            <img src="/src/assets/logo.svg" alt="" className="" />
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: Please give some meaningful name

File: src/components/header/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            <img src="/src/assets/logo+.svg" alt="" />
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+          </div>
+          <div className="hidden md:block">
+            <ul className="flex gap-24 md:gap-5 xl:gap-24  lg:gap-16  items-center font-semibold text-gray-400 ">
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: not formatted

File: src/components/header/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+              <li>
+                <a
+                  href="#"
+                  className={`text-gray-400 ${
+                    activeLink === "WORK" ? "text-gray-700" : ""
+                  }`}
+                  onClick={() => setActive("WORK")}
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: Please do not use the arrow function in the template. instead, create one function and use the same.

File: src/components/header/Header.jsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+        <div className="flex justify-between  items-center py-4">
+          <div className="flex  gap-2 md:gap-8  justify-center items-center">
+            <img src="/src/assets/logo.svg" alt="" className="" />
+            <img src="/src/assets/logo+.svg" alt="" />
+          </div>
+          <div className="hidden md:block">
+            <ul className="flex gap-24 md:gap-5 xl:gap-24  lg:gap-16  items-center font-semibold text-gray-400 ">
+              <li>
+                <a
+                  href="#"
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: Why hash link?

File: src/components/header/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                  className={`text-gray-400 ${
+                    activeLink === "WORK" ? "text-gray-700" : ""
+                  }`}
+                  onClick={() => setActive("WORK")}
+                >
+                  WORK
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: move this in to a constant

File: src/components/header/Header.jsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+                </a>
+              </li>
+              <li>
+                <a
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+                  className={`text-gray-400 ${
+                    activeLink === "SERVICES" ? "text-gray-700" : ""
+                  }`}
+                  onClick={() => setActive("SERVICES")}
+                >
+                  SERVICES
+                </a>
+              </li>
+              <li>
+                <a
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+                  className={`text-gray-400 ${
+                    activeLink === "ABOUT US" ? "text-gray-700" : ""
+                  }`}
+                  onClick={() => setActive("ABOUT US")}
+                >
+                  ABOUT US
+                </a>
+              </li>
+              <li>
+                <a
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+                  href="#"
+                  className={`text-gray-400 ${
+                    activeLink === "LABS" ? "text-gray-700" : ""
+                  }`}
+                  onClick={() => setActive("LABS")}
+                >
+                  LABS
+                </a>
+              </li>
+              <button className="bg-[#0A66EC] text-white rounded-full opacity-1 px-8 py-3 text-opacity-90 hover:bg-gradient-to-r from-blue-600 from-30%  to-pink-400">
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: why link inside a button?

File: src/components/header/Header.jsx [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+        </div>
+        <div
+          className={`space-y-4 px-4 mt-16 py-7 transition-all duration-300 ease-in  ${
+            isMenuOpen ? "block fixed top-0 right-0 left-0" : "hidden"
+          }`}
+        >
+          <ul className="flex flex-col gap-4 mt-8">
+            <li>
+              <a
+                href="#"
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+                  activeLink === "WORK" ? "text-gray-700" : ""
+                }`}
+                onClick={() => setActive("WORK")}
+              >
+                WORK
+              </a>
+            </li>
+            <li>
+              <a
+                href="#"
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+                  activeLink === "SERVICES" ? "text-gray-700" : ""
+                }`}
+                onClick={() => setActive("SERVICES")}
+              >
+                SERVICES
+              </a>
+            </li>
+            <li>
+              <a
+                href="#"
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

File: src/components/header/Header.jsx [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+                className={`block font-semibold text-gray-400 ${
+                  activeLink === "ABOUT US" ? "text-gray-700" : ""
+                }`}
+                onClick={() => setActive("ABOUT US")}
+              >
+                ABOUT US
+              </a>
+            </li>
+            <li>
+              <a
```
Comment (ram-primathon) [inline src/components/header/Header.jsx]: same

**Review by ram-primathon — COMMENTED** [review]
This is a good PR due to attached video. I don't have to run this code locally to see the outcomes.

File: src/components/header/Header.jsx (hunk 4 above)
Comment (Himanshushakyaa) [reply src/components/header/Header.jsx]: given the meaning full name

File: src/components/header/Header.jsx (hunk 7 above)
Comment (Himanshushakyaa) [reply src/components/header/Header.jsx]: complete change  and corred this

File: src/components/header/Header.jsx (hunk 13 above)
Comment (Himanshushakyaa) [reply src/components/header/Header.jsx]: all hash deleted and complete change and correct this

File: src/components/header/Header.jsx [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, []);
+
+  return (
+    <div className="fixed w-full">
+      <div className="flex justify-between p-6 md:p-0 md:py-5 z-50 md:px-16 lg:px-20 xl:px-28 2xl:px-36 items-center bg-white bg-opacity-50 backdrop-blur-md">
+          <div className="flex justify-center gap-x-3.5">
+            <img src="/src/assets/brand-logo.svg" className="w-4 h-6 2xl:w-6 2xl:h-8" />
```
Comment (ram-primathon) [inline src/components/header/Header.jsx:30]: use one icon

File: src/components/header/Header.jsx [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+          <div className={`hidden md:flex justify-center items-center gap-4`}>
+                        {headerJSON.map((link, index) => (
+                            <a
+                                key={`header-${index}`}
+                                className={`md:text-xs lg:text-sm xl:text-base text-[#2F3750] font-normal hover:font-semibold cursor-pointer lg:w-24 2xl:w-28`}>
+                                {link.title}
+                            </a>
+                        ))}
+                        <a
+                       className="hidden md:flex md:rounded-3xl xl:rounded-[32px] bg-[#0A66EC] transition duration-500 ease-in-out font-base focus:font-semibold transform focus:scale-100 hover:bg-gradient-to-r hover:from-[#0A66EC] hover:to-[#F46997] md:text-[10px] lg:text-xs xl:text-sm text-white px-6 py-3.5 ">
```
Comment (ram-primathon) [inline src/components/header/Header.jsx:42]: remove hardcoded color

File: src/components/header/Header.jsx [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+      <div className="flex justify-between p-6 md:p-0 md:py-5 z-50 md:px-16 lg:px-20 xl:px-28 2xl:px-36 items-center bg-white bg-opacity-50 backdrop-blur-md">
+          <div className="flex justify-center gap-x-3.5">
+            <img src="/src/assets/brand-logo.svg" className="w-4 h-6 2xl:w-6 2xl:h-8" />
+            <img src="/src/assets/yume-labs-logo.svg" className="w-36 md:w-30 md:h-5 xl:w-44 2xl:h-6 mt-1"/>
+          </div>
+          <div className={`hidden md:flex justify-center items-center gap-4`}>
+                        {headerJSON.map((link, index) => (
+                            <a
+                                key={`header-${index}`}
+                                className={`md:text-xs lg:text-sm xl:text-base text-[#2F3750] font-normal hover:font-semibold cursor-pointer lg:w-24 2xl:w-28`}>
```
Comment (ram-primathon) [inline src/components/header/Header.jsx:37]:
```suggestion
                                className="md:text-xs lg:text-sm xl:text-base text-[#2F3750] font-normal hover:font-semibold cursor-pointer lg:w-24 2xl:w-28">
```

File: src/components/header/Header.jsx [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+          <div className="md:hidden">
+            <button
+              onClick={toggleMenu}
+              className="text-white focus:outline-none transform transition-transform duration-500 ease-in-out"
+              style={{
+                transform: isMenuOpen ? "rotate(90deg)" : "rotate(0deg)",
+              }}
+            >
+              {isMenuOpen ? (
+                <span className="text-3xl  text-[#253F9C]">&#10005;</span>
```
Comment (ram-primathon) [inline src/components/header/Header.jsx:57]: Don't use this

File: src/data/header.json [hunk 21]
```diff
@@ -0,0 +1,22 @@
+[
+    {
+        "title": "WORK",
+        "link": "#work"
```
Comment (ram-primathon) [inline src/data/header.json:4]:
```suggestion
        "link": "/work"
```

**Review by ram-primathon — CHANGES_REQUESTED** [review]

**Pull request closed without merging**